pkg/backend/httpstate/backend.go

```diff
@@ -1416,10 +1416,13 @@ func (b *cloudBackend) runEngineAction(
 	close(engineEvents)
 	if snapshotManager != nil {
 		err = snapshotManager.Close()
-		// Historically we ignored this error (using IgnoreClose so it would log to the V11 log).
-		// To minimize the immediate blast radius of this to start with we're just going to write an error to the user.
+		// This error could be a snapshot write failure. While the operation as a
+		// whole may still have succeeded, with the write failing afterwards, we
+		// still raise this as an operation error since if it's a snapshot integrity
```

Comment:
What if writing the snapshot fails instead of verifying its integrity?  I'm think it might still be fine to fail the operation here in that case, but the comment is very vague about that case, so I'm curious if the intention is the error out here anyway, or we just error out for all types of errors because it's convenient?

Reply:
Yeh definitely good to fail the operation in that case. They'll have missing state data in that case, needs fixing asap.

Reply:
I'll improve the comments though -- good feedback :+1: